prometheus/value_test.go

```diff
@@ -108,3 +110,132 @@ func TestNewConstMetricWithCreatedTimestamp(t *testing.T) {
+
+func TestMakeLabelPairs(t *testing.T) {
+	tests := []struct {
+		name        string
+		desc        *Desc
+		labelValues []string
+		want        []*dto.LabelPair
+	}{
+		{
+			name:        "no labels",
+			desc:        NewDesc("metric-1", "", nil, nil),
+			labelValues: nil,
+			want:        nil,
+		},
+		{
+			name: "only constant labels",
+			desc: NewDesc("metric-1", "", nil, map[string]string{
+				"label-1": "1",
+				"label-2": "2",
+				"label-3": "3",
+			}),
+			labelValues: nil,
+			want: []*dto.LabelPair{
+				{Name: proto.String("label-1"), Value: proto.String("1")},
+				{Name: proto.String("label-2"), Value: proto.String("2")},
+				{Name: proto.String("label-3"), Value: proto.String("3")},
+			},
+		},
+		{
+			name:        "only variable labels",
+			desc:        NewDesc("metric-1", "", []string{"var-label-1", "var-label-2", "var-label-3"}, nil),
+			labelValues: []string{"1", "2", "3"},
+			want: []*dto.LabelPair{
+				{Name: proto.String("var-label-1"), Value: proto.String("1")},
+				{Name: proto.String("var-label-2"), Value: proto.String("2")},
+				{Name: proto.String("var-label-3"), Value: proto.String("3")},
+			},
+		},
+		{
+			name: "variable and const labels",
+			desc: NewDesc("metric-1", "", []string{"var-label-1", "var-label-2", "var-label-3"}, map[string]string{
+				"label-1": "1",
+				"label-2": "2",
+				"label-3": "3",
+			}),
+			labelValues: []string{"1", "2", "3"},
+			want: []*dto.LabelPair{
+				{Name: proto.String("label-1"), Value: proto.String("1")},
+				{Name: proto.String("label-2"), Value: proto.String("2")},
+				{Name: proto.String("label-3"), Value: proto.String("3")},
+				{Name: proto.String("var-label-1"), Value: proto.String("1")},
+				{Name: proto.String("var-label-2"), Value: proto.String("2")},
+				{Name: proto.String("var-label-3"), Value: proto.String("3")},
+			},
+		},
+		{
+			name: "unsorted variable and const labels are sorted",
+			desc: NewDesc("metric-1", "", []string{"var-label-3", "var-label-2", "var-label-1"}, map[string]string{
+				"label-3": "3",
+				"label-2": "2",
+				"label-1": "1",
+			}),
+			labelValues: []string{"3", "2", "1"},
+			want: []*dto.LabelPair{
+				{Name: proto.String("label-1"), Value: proto.String("1")},
+				{Name: proto.String("label-2"), Value: proto.String("2")},
+				{Name: proto.String("label-3"), Value: proto.String("3")},
+				{Name: proto.String("var-label-1"), Value: proto.String("1")},
+				{Name: proto.String("var-label-2"), Value: proto.String("2")},
+				{Name: proto.String("var-label-3"), Value: proto.String("3")},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := MakeLabelPairs(tt.desc, tt.labelValues); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("%v != %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Benchmark_MakeLabelPairs(b *testing.B) {
+	benchFunc := func(desc *Desc, variableLabelValues []string) {
+		MakeLabelPairs(desc, variableLabelValues)
+	}
+
+	benchmarks := []struct {
```

Comment:
nit: Do you mind doing inlined loop syntax? No point in having extra variable here, only one use.

```
for _, bm := []struct{}{....} { b.Run...}
```